torchtune/training/precision.py

```diff
@@ -50,6 +52,7 @@ def verify_bf16_support() -> bool:
             - CUDA compute capability >= 8
         - NCCL is available and version >= 2.10
         - MPS is available and torch was built with MPS
+        - NPU is available and supports bf16
```

Comment:
nit: this is a bit redundant. Do we know the exact requirements for bf16 support on NPUs?